### PR TITLE
typescript for projectPropContext

### PR DESCRIPTION
**Pull request by sunilsabatp**
Fixes 
typescript issues

### CODE REVIEW

File: src/features/common/Layout/ProjectPropsContext.tsx [hunk 1]
```diff
@@ -1,124 +1,246 @@
...
   plantLocationsLoaded: false,
-  setPlantLocationsLoaded: (value: boolean) => {},
-});
+  setPlantLocationsLoaded: () => {},
+  isMobile: false,
+  setIsMobile: () => {},
+};
+
+export const ProjectPropsContext =
+  React.createContext<ProjectContextInterface>(defaultState);
```
Comment (norbertschuler) [inline src/features/common/Layout/ProjectPropsContext.tsx:183]: In my VSC I still get displayed this error:

```
Argument of type '{ project: {} | null; setProject: () => void; projects: never[]; setProjects: () => void; plantLocations: never[]; setPlantLocations: () => void; showSingleProject: boolean; setShowSingleProject: () => void; ... 66 more ...; setIsMobile: () => void; }' is not assignable to parameter of type 'ProjectContextInterface'.
  Types of property 'project' are incompatible.
    Type '{} | null' is not assignable to type 'Project | null'.
      Type '{}' is missing the following properties from type 'Project': id, _scope, allowDonations, classification, and 44 more.ts(2345)
```
I am not yet experienced with Typescript typing, so maybe you have an idea about this. I think I used the default settings for Typescript checks in Visual Studio Code.

Comment (norbertschuler) [reply src/features/common/Layout/ProjectPropsContext.tsx:183]: I changes the types and default values. Please have a look. If this is wrong, please revert my commit.

Comment (norbertschuler) [reply src/features/common/Layout/ProjectPropsContext.tsx:183]: I again have problems as you used uppercase characters for branch names `Feature/typescript_projectPropContext`. Can we please not used uppercase characters in branch names?

Comment (sunilsabatp) [reply src/features/common/Layout/ProjectPropsContext.tsx:183]: I am not sure. but, whenever I create branch using lowercase character, I am unable to push the changes into remote repo

**Comment by mohitb35** [comment]
Can we close this as #1655 corrects the branch name?

**Comment by mohitb35** [comment]
Closed and working with #1655

**Pull request closed without merging**